Hw01

nekobehemoth opened the pull request.

ushakofuiria committed:
Added initial implementation of CustomList

ushakofuiria committed:
Add rest of Basic Implementation and unit tests

ushakofuiria committed:
Add generics, implement a few more method,  fix bugs, improve tests

ushakofuiria committed:
Add perfomance testing and report

ushakofuiria committed:
The last fixes

ushakofuiria committed:
change the package name

gneginskiy commented on src/test/java/org/nekobehemoth/gehtsofttraining/TestTiming.java:
```diff
@@ -0,0 +1,72 @@
+    }
+
+    @Override
+    public void afterAll(ExtensionContext extensionContext) throws Exception {
```
please change to appropriate annotations (@AfterAll, @BeforeAll or smth like this),  for the method: it's more common/idiomatic/obvious way than using  inheritance with callback methods

nekobehemoth replied:
Hello @gneginskiy, moved to PerformanceTest class and changed to annotation
![image](https://github.com/user-attachments/assets/8a226788-654d-43ed-be78-5020057bd160)

gneginskiy commented:
nope, it's broken 👀 . please fix.
<img width="1721" alt="image" src="https://github.com/user-attachments/assets/257ea185-10b8-4e3d-a091-843bfab11dc2" />

ushakofuiria committed:
fix add method, and change afterAll to annotation

nekobehemoth commented:
Hello Grigorii, 

Fixed by returning a copy of the Array, but not new object 

![Fixed](https://github.com/user-attachments/assets/a3446592-9bcf-4090-ac8c-0c39cb1c0972)

Also added the test for it
![image](https://github.com/user-attachments/assets/f50f42d3-25d4-4441-b210-e28fee2bb4cd)

nekobehemoth committed:
Delete src/test/java/org/nekobehemoth/hw02/fibonacci/FibonacciAlgorithmsTest.java

nekobehemoth committed:
Delete src/main/java/org/nekobehemoth/hw02/fibonacci/FibonacciAlgorithms.java
